File: isma-intg-lib/isma-intg-lib-rkfehlberg/src/main/java/ru/nstu/isma/intg/lib/rungeKutta/rungeKuttaFehlberg/internal/RkFehlbergStage6Calculator.java
package ru.nstu.isma.intg.lib.rungeKutta.rungeKuttaFehlberg.internal;

import ru.nstu.isma.intg.api.methods.*;

public class RkFehlbergStage6Calculator extends StageCalculator {
	@Override
	public double yk(final double step, final double y, final double f, final double[] stages) {
		return y - 0.2962962962 * stages[0] + 2.0 * stages[1] - 1.3816764133 * stages[2] + 0.4529727096 * stages[3]
				- 0.275 * stages[4];
	}

	@Override
	public double k(final double step, final double y, final double f, final double[] stages, final double stagesY,
			final double stagesF) {
		return step * stagesF;
	}
}

File: isma-intg-lib/isma-intg-lib-rkfehlberg/src/main/kotlin/ru/nstu/isma/intg/lib/rungeKutta/rungeKuttaFehlberg/internal/RkFehlbergStage6Calculator.kt
package ru.nstu.isma.intg.lib.rungeKutta.rungeKuttaFehlberg.internal

import ru.nstu.isma.intg.api.methods.StageCalculator

class RkFehlbergStage6Calculator : StageCalculator() {
    override fun yk(step: Double, y: Double, f: Double, stages: DoubleArray): Double {
        return (y - 0.2962962962 * stages[0] + 2.0 * stages[1] - 1.3816764133 * stages[2] + 0.4529727096 * stages[3]
                - 0.275 * stages[4])
    }

    override fun k(
        step: Double, y: Double, f: Double, stages: DoubleArray, stagesY: Double,
        stagesF: Double
    ): Double {
        return step * stagesF
    }
}
